### PR TITLE
add bwameth modules

### PR DESCRIPTION
**Pull request by phue**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --wt 2 --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --wt 2 --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --wt 2 --keep-workflow-wd`

### CODE REVIEW

**Comment by phue** [comment]
:man_shrugging: 

```
Caused by:
  Failed to create Conda environment
  command: conda create --mkdir --yes --quiet --prefix /home/runner/pytest_workflow_x5yy98ht/Run_bwameth_single-end_test_workflow/work/conda/env-2f0ca6b52ef420f0984503f86281f07d bioconda::bwameth=0.2.2
  status : 1
  message:
    InvalidArchiveError("Error with archive /home/runner/conda_pkgs_dir/python-3.9.1-hffdb5ce_5_cpython.tar.bz2.  You probably need to delete and re-download or re-create this file.  Message from libarchive was:\n\nFailed to create dir 'lib'")
    InvalidArchiveError("Error with archive /home/runner/conda_pkgs_dir/perl-5.32.0-h36c2ea0_0.tar.bz2.  You probably need to delete and re-download or re-create this file.  Message from libarchive was:\n\nFailed to create dir 'lib'")
```

File: software/bwameth/align/main.nf [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    script:
+    def software   = getSoftwareName(task.process)
+    def prefix     = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    def read_group = meta.read_group ? "-R ${meta.read_group}" : ""
+    """
+    bwameth.py \\
+        $options.args \\
+        $read_group \\
+        -t $task.cpus \\
+        --reference ${index}/genome.fa \\
```
Comment (drpatelh) [inline software/bwameth/align/main.nf:38]: We aren't guaranteed for the index to be named `genome.fa`. [This](https://github.com/nf-core/modules/blob/01c0d335ca57b1b9298a7acd643116917953804a/software/bwa/mem/main.nf#L34) is a more robust solution.

File: software/bwameth/index/main.nf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        container "https://depot.galaxyproject.org/singularity/bwameth:0.20--py35_0"
+    } else {
+        container "quay.io/biocontainers/bwameth:0.20--py35_0"
+    }
+
+    input:
+    path fasta, stageAs: "index/genome.fa"
+
+    output:
+    path "index"        , emit: index
```
Comment (drpatelh) [inline software/bwameth/index/main.nf:25]:
```suggestion
    path "bwameth"        , emit: index
```

Comment (drpatelh) [reply software/bwameth/index/main.nf:25]: Given the way we are creating this for other indexing modules be great if we can be consistent. See [here](https://github.com/nf-core/modules/blob/01c0d335ca57b1b9298a7acd643116917953804a/software/bwa/index/main.nf#L25)

File: software/bwameth/index/main.nf [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    conda (params.enable_conda ? "bioconda::bwameth=0.2.2" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/bwameth:0.20--py35_0"
+    } else {
+        container "quay.io/biocontainers/bwameth:0.20--py35_0"
+    }
+
+    input:
+    path fasta, stageAs: "index/genome.fa"
```
Comment (drpatelh) [inline software/bwameth/index/main.nf:22]:
```suggestion
    path fasta, stageAs: "bwameth/*"
```

File: software/bwameth/index/main.nf [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    output:
+    path "index"        , emit: index
+    path "*.version.txt", emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    """
+    bwameth.py index $fasta
```
Comment (drpatelh) [inline software/bwameth/index/main.nf:31]:
```suggestion
    bwameth.py index bwameth/${fasta}
```

**Review by drpatelh — CHANGES_REQUESTED** [review]
Thanks! Some minor changes @phue.

File: software/bwameth/align/main.nf (hunk 1 above)
Comment (phue) [reply software/bwameth/align/main.nf:38]: Alright, alright :see_no_evil: 
I'm not a big fan of it (looks a bit hacky), so I thought it's enough to ensure the `bwameth_index` module always uses `genome.fa` as filename. It's also the case for iGenomes.

But I don't have strong opinions on it, so here we go :)

**Review by drpatelh — APPROVED** [review]
Thanks @phue !

**Pull request merged**